graphql/src/query.rs

```diff
@@ -292,107 +291,42 @@ impl Query {
-        // have performance issues. The query would roughly be:
-        //     delete from comment where user_id and comment_id
-        //     returning parent_type, parent_id
 
-        let db = context.data::<DataLoader<Database>>()?.loader();
-
-        let comment = db.comment_by_id(&input.comment_id).await?;
-
-        if comment.posted_by.id.0 != user.id.to_string() {
-            return Err("User attempted to delete another user's comment".into());
-        }
-
-        db.delete_comment(&input.comment_id).await?;
-
-        // We return the parent object, for GraphCache interop
-        comment.parent(context).await
-    }
-
-    /// Post a new comment on a given object
-    #[graphql(guard = "AuthGuard")]
-    async fn post_comment(
-        &self,
-        context: &Context<'_>,
-        input: PostCommentInput,
-    ) -> FieldResult<CommentParent> {
-        let user = context
-            .data_opt::<UserInfo>()
-            .ok_or_else(|| anyhow::format_err!("User is not signed in"))?;
-
-        let db = context.data::<DataLoader<Database>>()?.loader();
-
-        db.insert_comment(
-            &user.id,
-            input.text_content,
-            &input.parent_id,
-            &input.parent_type,
-            &input.comment_type,
-        )
-        .await?;
-
-        // We return the parent object, for GraphCache interop
-        input.parent_type.resolve(db, &input.parent_id).await
-    }
-
-    /// Mutation for adding/changing contributor attributions
-    #[graphql(
-        guard = "GroupGuard::new(UserGroup::Editors).or(GroupGuard::new(UserGroup::Contributors))"
-    )]
-    async fn update_contributor_attribution(
-        &self,
-        context: &Context<'_>,
-        contribution: UpdateContributorAttribution,
-    ) -> FieldResult<Uuid> {
+    /// Gets all bookmarks for the currently authenticated user.
+    async fn get_bookmarks(&self, context: &Context<'_>) -> FieldResult<Option<Vec<Uuid>>> {
+        // let user = context
+        //     .data_opt::<UserInfo>()
+        //     .ok_or_else(|| anyhow::format_err!("User is not signed in"))?;
         Ok(context
             .data::<DataLoader<Database>>()?
             .loader()
-            .update_contributor_attribution(contribution)
+            .get_bookmarks(
+                Uuid::parse_str("910a22ac-21cd-4398-b656-1039931de29f")
+                    .unwrap_or_else(|_| panic!("Failed to parse UUID")),
+            )
             .await?)
     }
 
-    ///Mutation for deleting contributor attributions
-    #[graphql(
-        guard = "GroupGuard::new(UserGroup::Editors).or(GroupGuard::new(UserGroup::Contributors))"
-    )]
-    async fn delete_contributor_attribution(
-        &self,
-        context: &Context<'_>,
-        contribution: DeleteContributorAttribution,
-    ) -> FieldResult<Uuid> {
+    /// Gets the short name of a document given its id
+    async fn get_doc_short_name(&self, context: &Context<'_>, doc_id: Uuid) -> FieldResult<String> {
```

Comment:
Like I said on the schema file, this should be a field resolver on the `AnnotatedDoc` type, not a top-level query.